components/sup/src/manager.rs

```diff
@@ -1785,19 +1785,38 @@ fn track_memory_stats() {
     // We'd like to track some memory stats, but these stats are cached and only refreshed
     // when the epoch is advanced. We manually advance it here to ensure our stats are
     // fresh.
-    jemalloc_ctl::epoch().unwrap();
+    jemalloc_ctl::epoch::mib().unwrap().advance().unwrap();
+
     MEMORY_STATS.with_label_values(&["active"])
-                .set(jemalloc_ctl::stats::active().unwrap().to_i64());
+                .set(jemalloc_ctl::stats::active::mib().unwrap()
+                                                       .read()
+                                                       .unwrap()
+                                                       .to_i64());
     MEMORY_STATS.with_label_values(&["allocated"])
-                .set(jemalloc_ctl::stats::allocated().unwrap().to_i64());
+                .set(jemalloc_ctl::stats::allocated::mib().unwrap()
+                                                          .read()
+                                                          .unwrap()
+                                                          .to_i64());
     MEMORY_STATS.with_label_values(&["mapped"])
-                .set(jemalloc_ctl::stats::mapped().unwrap().to_i64());
+                .set(jemalloc_ctl::stats::mapped::mib().unwrap()
+                                                       .read()
+                                                       .unwrap()
+                                                       .to_i64());
     MEMORY_STATS.with_label_values(&["metadata"])
-                .set(jemalloc_ctl::stats::metadata().unwrap().to_i64());
+                .set(jemalloc_ctl::stats::metadata::mib().unwrap()
+                                                         .read()
+                                                         .unwrap()
+                                                         .to_i64());
     MEMORY_STATS.with_label_values(&["resident"])
-                .set(jemalloc_ctl::stats::resident().unwrap().to_i64());
+                .set(jemalloc_ctl::stats::resident::mib().unwrap()
+                                                         .read()
+                                                         .unwrap()
+                                                         .to_i64());
     MEMORY_STATS.with_label_values(&["retained"])
-                .set(jemalloc_ctl::stats::retained().unwrap().to_i64());
+                .set(jemalloc_ctl::stats::retained::mib().unwrap()
+                                                         .read()
+                                                         .unwrap()
+                                                         .to_i64());
```

Comment:
Why did all these change?

I'm a bit uneasy with all the unwraps... can we be certain they're safe to do?

At the _very_ least, they should be `expect` calls, so we can get additional context if one of them fires.

Reply:
The version of jemalloc got bumped in the dep chain... in general seems fine to pick up an updated version, the only change is that the stats is under a mib namespace and needed another unwrap than what was already there.